core/controllers/domain_objects_validator.py

```
@@ -43,7 +43,7 @@
     stats_domain,
 )
 
-from typing import Dict, Mapping, Optional, Union
+from typing import Dict, Mapping, Optional
```

Ruff (0.15.17)

[warning] Import from `collections.abc` instead: `Mapping`

Import from `collections.abc`

(UP035)